### PR TITLE
DOC: Mention negative masses for center of mass

### LINKED ISSUE #6667: center_of_mass accepts negative values
center_of_mass accepts negative values without warning, which leads to very unexpected behavior, e.g a center of mass that is outside the dimensions of the input array:

    from scipy.ndimage.measurements import center_of_mass
    center_of_mass(np.array([1,3,-5]))

returns ```(7.0,)```. Of course, a negative mass is not a very sensible input, but this should be at least mentioned in the documentation. Probably even better would be to give a warning or only consider positive values, as is done when negative labels are specified.

### PR DESCRIPTION
**Pull request by gfyoung**
Explicitly mention that data inputs for the center of mass function can be positive or negative.

Closes #6667.

### CODE REVIEW

**Comment by andyfaff** [comment]
I'm fine with the UserWarning, but are we absolutely sure that there are no use cases for negative values? Whilst there may not be immediate uses in the field of image processing, I could imagine that negative values might occur in some other scientific usage of this function.

**Comment by gfyoung** [comment]
Perhaps we could pose that question to the mailing list?  If there are applications, the least I can do is update documentation to say we accept both positive and negative values.

**Comment by gfyoung** [comment]
In light of the email that I sent out, this PR will just be a minor documentation update.

**Comment by person142** [comment]
Should probably also be explicit handling of this case:

```python
>>> from scipy.ndimage import center_of_mass
>>> import numpy as np
>>> center_of_mass(np.array([-1, 1]))
/home/wil00108/Projects/scipy-dev/python2-scipy-dev/scipy/scipy/ndimage/measurements.py:1284: RuntimeWarning: divide by zero encountered in double_scalars
  for dir in range(input.ndim)]
(inf,)
```

Also an example with negative masses might be nice?

**Comment by jjmetzger** [comment]
I think a function called center_of_mass should at least warn about negative values, since the name suggests that the mass should be always >0 and the center of mass within the array.

**Comment by person142** [comment]
Link to mailing list discussion:

https://mail.scipy.org/pipermail/scipy-dev/2016-October/021573.html

**Comment by gfyoung** [comment]
@jjmetzger : A warning is unnecessary if we say that negative values can be accepted in the documentation.  Given that there are applications, there is really nothing to warn about IMO.

File: scipy/ndimage/measurements.py [hunk 1]
```diff
@@ -1242,7 +1242,8 @@ def center_of_mass(input, labels=None, index=None):
     Parameters
     ----------
     input : ndarray
-        Data from which to calculate center-of-mass.
+        Data from which to calculate center-of-mass. The data can either
+        be positive or negative.
```
Comment (gfyoung) [inline scipy/ndimage/measurements.py]: Nitpick away!  Documentation ought to be as clear as possible. :smile:

**Comment by jjmetzger** [comment]
ok, I guess the updated documentation is enough. thanks!

**Comment by gfyoung** [comment]
@person142 , @jjmetzger : PTAL

**Comment by person142** [comment]
ISTM that `inf` is the "correct" solution for a system like `[-1, 1]`. (At least when I think of it as two masses, one of which wants to fall up.) From that perspective, I would like for the function to return `(inf,)` in that case without any warnings. (And a tuple of `inf`'s for higher dimensions representing the point at infinity.)

But maybe I'm thinking through this incorrectly, so let's see if someone wants to educate me. :-)

**Comment by gfyoung** [comment]
@person142 : Fair enough, but the warning is coming from `numpy`, so it's out of our control to some degree, unless you want to "filter out" warnings but that seems risky IMO.

**Comment by person142** [comment]
I'm suggesting detecting which elements of `normalizer` are exactly zero and handling those cases separately.

**Comment by gfyoung** [comment]
@person142 : Ah, I see.  Perhaps we could save that for a follow-up since this PR was focused on just accepting negative values for masses?

**Comment by person142** [comment]
Thanks @gfyoung!

**Pull request merged**